### PR TITLE
fixing/improving and cleanup tablet inputs

### PR DESCRIPTION
**Pull request by druiz17**
This PR attempts to make the stylus and laser more reliable. This PR also moves the function the send mouse event to it own util file to prevent duplicate code in multiple files.

### CODE REVIEW

**Comment by druiz17** [comment]
Test plan
- use the laser and stylus on tablet to see if it improves the behavior of what is currently in master.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3458/HighFidelity-Beta-PR11408-9c66a25d837268615ce7d6cbf670377f438452eb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3458/HighFidelity-Beta-PR11408-9c66a25d837268615ce7d6cbf670377f438452eb.exe

**Comment by anshuman64** [comment]
@druiz17 This is not working as well as what you showed me on your computer. Let's double check tomorrow...

**Comment by ZappoMan** [comment]
Can you explain more what you think this is going to address? what specific design issue/flaw in the original code is addressed by this change? I see you DRY up the code, but how does that fix what bug?

**Comment by druiz17** [comment]
The major issue is that the stylus and lasers for each hand wants the touch focus of the tablet. We do not handle requesting focus and releasing touch focus correctly with causes the tablet to be unresponsive at time. This is the major issue that I am trying to address. I am also moving duplicate touchEvent code that exists in more than one file into a util file for easier maintenance down the line.

File: scripts/system/controllers/controllerModules/tabletStylusInput.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ Script.include("/~/system/libraries/controllerDispatcherUtils.js");
 Script.include("/~/system/libraries/controllers.js");
 
 (function() {
-
+    var touchEvent = Script.require("/~/system/libraries/touchEventUtils.js");
```
Comment (hyperlogic) [inline scripts/system/controllers/controllerModules/tabletStylusInput.js]: It isn't in our docs but the usual JS convention for modules is to use upper case.  Also, IMHO, it should have the same name as the library file, So this should instead be something like:

    var TouchEventUtils = Script.require("/~/system/libraries/touchEventUtils.js");

File: scripts/system/controllers/controllerModules/overlayLaserInput.js [hunk 2]
```diff
@@ -16,6 +16,7 @@ Script.include("/~/system/libraries/controllerDispatcherUtils.js");
 Script.include("/~/system/libraries/controllers.js");
 
 (function() {
+    var touchEvent = Script.require("/~/system/libraries/touchEventUtils.js");
```
Comment (hyperlogic) [inline scripts/system/controllers/controllerModules/overlayLaserInput.js]: Same here:

    var TouchEventUtils = Script.require("/~/system/libraries/touchEventUtils.js");

File: scripts/system/controllers/controllerModules/tabletStylusInput.js (hunk 1 above)
Comment (hyperlogic) [reply scripts/system/controllers/controllerModules/tabletStylusInput.js]: Looking at some of node's internal modules, maybe I'm wrong here, JS conventions seem to change very frequntly... anyhow this is just like my opinion man. feel free to ignore it if you feel strongly the other way.

File: scripts/system/controllers/controllerModules/overlayLaserInput.js [hunk 3]
```diff
@@ -277,16 +98,11 @@ Script.include("/~/system/libraries/controllers.js");
     function OverlayLaserInput(hand) {
         this.hand = hand;
         this.active = false;
-        this.previousLaserClikcedTarget = false;
```
Comment (hyperlogic) [inline scripts/system/controllers/controllerModules/overlayLaserInput.js]: lol JavaSrcipt

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3474/HighFidelity-Beta-PR11408-2ce6dcf89dbb8a2368c1a9fcd486b15384e3a55d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3474/HighFidelity-Beta-PR11408-2ce6dcf89dbb8a2368c1a9fcd486b15384e3a55d.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3482/HighFidelity-Beta-PR11408-c20c5ee2b5e812c9c4c8322945abec8ec99b6788.dmg

**Pull request merged**